### DISCUSSION TITLE
(new) problem with Pico+PI 2B <<-- goes in boot loop.

### DISCUSSION THREAD

**Discussion by dhstsw** [probonopd/MiniDexed#451] [General]
Hi all,
I've been running minidexed so far without any problems.
It has a Pico connected to handle midi comunications; USB goes into the pico, then (beside the data pins) pin 39 of the pico (VBUS) is connected to the 2B (to a 5V) and then there is another wire going from one of the GND of the Pico to one off the GND of the 2B.
There's also an HD44780 as display.

It worked ok until some time ago (i noticed it a couple months ago even if i'm writing about it only now).

Now, pluggin the usb to the pic seems to send the 2B in a boot loop and nothing happens.
If i plug the USB directly to the 2B it boots up no problems (but i don't have midi anymore).

Tried to use pin 39 in the pico (VSYS), same result.

Any ideas?

Thanks.

**Comment by diyelectromusic** [comment]
Have you measured the voltage on the 5V line from the Pico to the Pi to confirm it is a good, solid 5V?  It might be worth checking with what is published of the Pi schematics that driving 5V directly like this is ok?  There is some discussion here: https://raspberrypi.stackexchange.com/questions/1617/how-do-i-supply-power-through-the-gpio

Is it worth trying things the other way round - powering the Pico (via VSYS) from the Pi 5V?

Kevin

**Comment by dhstsw** [comment]
> Have you measured the voltage on the 5V line from the Pico to the Pi to confirm it is a good, solid 5V?

It's 5.12v both on VBUS and VSYS.

> Is it worth trying things the other way round - powering the Pico (via VSYS) from the Pi 5V?

That would defeat the whole meaning of all (having a single usb cable for powering up the whole thing *and* USB Midi).

I'll check that link.

Thanks.

**Reply by diyelectromusic** [reply]
Ah yes of course :)

**Reply by dhstsw** [reply]
So, i ended up connecting the VBUS and GND out from the Pico to the test points (PP7 (VCC)) and PP3 (GND)) on the 2B.

https://forums.raspberrypi.com//viewtopic.php?f=28&t=84874

Now it works perfectly again.

Thanks for putting me on the right track :)

C.

**Reply by diyelectromusic** [reply]
Great!  Glad you've got it going again.  I must admit though, I'm really not sure why using the TPs is different to the GPIO 5V/GND... but I guess that is part of the issue of RPi only publishing part schematics and there isn't a lot of info online anymore for the Pi 2.

Anyway, as I say glad you've found a solution! :)

Kevin

**Reply by dhstsw** [reply]
From the infos i gathered from the links above:

-powering using the +5 / GND in the GPIO gives "no backward protection" (wich is given powering from the microusb);
-Using the testpoints, instead, "is functionally the same as powering the PI through the microUSB", so, i guess, the protection is not bypassed.

Cheers.

**Reply by diyelectromusic** [reply]
I saw those comments, but that is what I mean about incomplete schematics as (if my reading of them is correct) the test points and GPIO pins are connected on the same side of the "protection" - i.e. not covered.  I guess in the physical layout things like smoothing capacitors will be closer to the USB end of things than the GPIO end of things, so maybe the test points have "closer smoothing"... but not being an electronics person, I don't know if that is significant or not.

Anyway, as I say, main thing is it is now working for you!

Kevin
